Export customize

dashea opened the pull request:
This is the bdcs-api part of https://github.com/weldr/bdcs/pull/147

This adds a "customizations" section to recipes, currently with two kickstart-inspired commands: hostname and sshkey. All customizations are optional. The API server translates the customizations into WriteFile directives and passes those to the bdcs export command.

clumens commented on bdcs-api.cabal:
```diff
@@ -31,7 +31,7 @@ library
     build-depends:      aeson >= 1.2.2.0 && < 1.3.0.0,
                         base >= 4.9 && < 5.0,
-                        bdcs >= 0.2.1,
+                        bdcs >= 0.2.2,
```
There have been a lot of builds of bdcs lately, so you'll need to double check which one you need here.

clumens commented on src/BDCS/API/V0.hs:
```diff
@@ -1629,6 +1639,47 @@ compose cfg@ServerConfig{..} ComposeBody{..} test | cbType `notElem` supportedOu
+                            (C8.length keyData)
+                            Nothing
+         in [WriteFile keyDir Nothing, WriteFile keyFile (Just keyData)]
+
```
It might make sense to put these into a separate file.  I imagine the set of customizations is going to grow, potentially to be pretty large, and this file is already unwieldy.

dashea commented:
Moved the RecipeCustomization types and functions to a separate module. I removed the aeson version bump from the first commit, since 4ceef0ec1c2903f3f471d78e4f11f69bdbf9779e makes it unnecessary, but kept the pkg-config dependency for libgit2-glib.

dashea commented:
Slight change to the tests, based on actually looking at the test results. Let's see how they come out this time.
